### DISCUSSION TITLE
Creating a 2d map using the ros2djs

### DISCUSSION THREAD

**Discussion by ussefdesouky** [RobotWebTools/rosbridge_suite#749] [General]
Hi, I'm trying to build a map through a web page in ROS2.

When I run the turtlebot3 pkg and run any mapping pkg (cartographer or slam_toolbox) achieving a web view of the created map, this error pops up!
[https://answers.ros.org/upfiles/16468351852897503.png](url)

**Comment by CCCCe** [comment]
> 

Have you addressed that? I got same trouble in recent days.

**Comment by ussefdesouky** [comment]
Yes, I have installed ROS1 (Noetic) and installed the ros1 bridge pkg, I think you can use ros2_web_bridge.

**Reply by CCCCe** [reply]
> Yes, I have installed ROS1 (Noetic) and installed the ros1 bridge pkg, I think you can use ros2_web_bridge.

Sorry, I can't get what  your mean. should I install ROS1 and ROS2 under ubuntu together? ROS2 for gather data, ROS1 for the bridge to web？can you give more details for me. Thanks a lot.
